app/Traits/AccessControlAPI.php

```diff
@@ -132,16 +133,16 @@ private function bulkAccessControl($api_key, $fileset_hash)
 
             $dbp_database = config('database.connections.dbp.database');
 
-            return AccessGroupKey::join(
+            return AccessGroupKey::join('user_keys', function ($join) use ($api_key) {
+                $join->on('user_keys.id', '=', 'access_group_api_keys.key_id')
+                    ->where('user_keys.key', $api_key);
+            })->join(
                 $dbp_database . '.access_group_filesets as acc_filesets',
                 function ($join) use ($fileset_hash) {
                     $join->on('access_group_api_keys.access_group_id', '=', 'acc_filesets.access_group_id')
                         ->where('hash_id', $fileset_hash);
                 }
-            )->join('user_keys', function ($join) use ($api_key) {
-                    $join->on('user_keys.id', '=', 'access_group_api_keys.key_id')
-                        ->where('user_keys.key', $api_key);
-            })->get();
+            )->select('user_keys.id')->get();
```

Comment:
We only need to select the id column because the method that calls this query will only validate if the amount rows is greater than 1, so, we don't need to select all columns.